src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -0,0 +1,620 @@
+                case 1:
+                    switch (binaryOperation.OperatorKind)
+                    {
+                        case BinaryOperatorKind.LessThan:
+                            shouldNegate = false;
+                            break;
+                        case BinaryOperatorKind.GreaterThanOrEqual:
+                            shouldNegate = true;
+                            break;
+                        default:
+                            return false;
+                    }
+                    break;
+                default:
+                    return false;
+            }
+
+            return true;
+        }
+
+        private static bool IsRightCountComparison(IBinaryOperation binaryOperation, out bool shouldNegate)
+        {
+            shouldNegate = false;
+
+            if (!TryGetZeroOrOneConstant(binaryOperation.LeftOperand, out int constant))
+            {
+                return false;
+            }
+
+            switch (constant)
+            {
+                case 0:
+                    switch (binaryOperation.OperatorKind)
+                    {
+                        case BinaryOperatorKind.Equals:
+                        case BinaryOperatorKind.GreaterThanOrEqual:
+                            shouldNegate = false;
+                            break;
+
+                        case BinaryOperatorKind.LessThan:
+                        case BinaryOperatorKind.NotEquals:
+                            shouldNegate = true;
+                            break;
+
+                        default:
+                            return false;
+                    }
+                    break;
+                case 1:
+                    switch (binaryOperation.OperatorKind)
+                    {
+                        case BinaryOperatorKind.LessThanOrEqual:
+                            shouldNegate = true;
+                            break;
+
+                        case BinaryOperatorKind.GreaterThan:
+                            shouldNegate = false;
+                            break;
+
+                        default:
+                            return false;
+                    }
+                    break;
+                default:
+                    return false;
+            }
+
+            return true;
+        }
+
+        private static bool TypeContainsVsibileProperty(OperationAnalysisContext context, ITypeSymbol type, string propertyName, SpecialType propertyType)
+            => TypeContainsVsibileProperty(context, type, propertyName, propertyType, propertyType);
+
+        private static bool TypeContainsVsibileProperty(OperationAnalysisContext context, ITypeSymbol type, string propertyName, SpecialType lowerBound, SpecialType upperBound)
+        {
+            if (type.GetMembers(propertyName).FirstOrDefault() is IPropertySymbol property)
+            {
+                return IsPropertyValidAndVisible(context, property, lowerBound, upperBound);
+            }
```

Comment:
Seems like the same code is used three times in this method. Can we extract to a static local function?